extensions/graphql/src/graphql.container.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/graphql
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  Binding,
+  Constructor,
+  Context,
+  createBindingFromClass,
+  filterByKey,
+  filterByServiceInterface,
+} from '@loopback/core';
+import debugFactory from 'debug';
+import {ContainerType, ResolverData} from 'type-graphql';
+import {GraphQLBindings, GraphQLTags} from './keys';
+
+const debug = debugFactory('loopback:graphql:container');
+
+/**
+ * Context for graphql resolver resolution
+ */
+export class GraphQLResolutionContext extends Context {
+  constructor(
+    parent: Context,
+    readonly resolverClass: Constructor<unknown>,
+    readonly resolverData: ResolverData<unknown>,
+  ) {
+    super(parent);
+    this.bind(GraphQLBindings.RESOLVER_DATA).to(resolverData);
+    this.bind(GraphQLBindings.RESOLVER_CLASS).to(resolverClass);
+  }
+}
+
+/**
+ * Implementation of `ContainerType` to plug into `type-graphql` as the IoC
+ * container
+ */
+export class LoopBackContainer implements ContainerType {
+  constructor(readonly ctx: Context) {}
+  get(
+    resolverClass: Constructor<unknown>,
+    resolverData: ResolverData<unknown>,
+  ) {
+    debug('Resolving a resolver %s', resolverClass.name, resolverData);
+    const resolutionCtx = new GraphQLResolutionContext(
+      this.ctx,
+      resolverClass,
+      resolverData,
+    );
+    const resolverBinding = createBindingFromClass(resolverClass, {
+      defaultNamespace: 'resolvers',
+    });
+    // Find resolver bindings that match the class
+    const bindings = this.ctx
+      .findByTag(GraphQLTags.RESOLVER)
+      .filter(filterByServiceInterface(resolverClass));
```

Comment:
It makes me wonder what is the performance cost of scanning all bindings whenever we want to instantiate a resolver class. 

I also don't understand why is this scan needed since we already have the binding key (`resolverBinding.key`) we want to resolve/instantiate? Cannot we use `ctx.get(resolverBinding.key)` directly? You can check whether the binding exists by calling `ctx.isBound(resolverBinding.key)`.

Calling `createBindingFromClass` could be a performance problem too, but I don't see any easy solution, so let's leave this part for later.

Reply:
`findByTag` uses tag index internally and the overhead is minimal. 

The idea here is to use existing bindings first before we add a new binding created from the class.